contracts/test/StakeTable.t.sol

```diff
@@ -242,21 +245,420 @@ contract StakeTable_register_Test is Test {
+        EdOnBN254.EdOnBN254Point memory emptySchnorrVK = EdOnBN254.EdOnBN254Point(0, 0);
+
+        // Step 6: generate a new blsVK
+        seed = "235";
+        (BN254.G2Point memory newBlsVK2,, BN254.G1Point memory newSig2) =
+            genClientWallet(exampleTokenCreator, seed);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit AbstractStakeTable.UpdatedConsensusKeys(exampleTokenCreator);
+        stakeTable.updateConsensusKeys(newBlsVK2, emptySchnorrVK, newSig2);
+
+        // Step 7: verify the update
+        node = stakeTable.lookupNode(exampleTokenCreator);
+        assertTrue(stakeTable._isEqualBlsKey(node.blsVK, newBlsVK2)); // same as current bls vk
+        assertTrue(EdOnBN254.isEqual(node.schnorrVK, schnorrVK)); // same as current schnorr vk
+        assertEq(node.balance, depositAmount); //same balance
+        assertEq(node.account, exampleTokenCreator); //same account
+
+        vm.stopPrank();
+    }
+
+    function test_lookupNodeAndLookupStake_fails() public {
+        address randomUser = makeAddr("randomUser");
+
+        // lookup the stake for an address that is not registered and verify the data is empty
+        uint256 stakeAmount = stakeTable.lookupStake(randomUser);
+        assertEq(stakeAmount, 0);
+
+        // lookup the node for an address that is not registered and verify the data is empty
+        AbstractStakeTable.Node memory node = stakeTable.lookupNode(randomUser);
+        assertEq(node.account, address(0));
+        assertEq(node.balance, 0);
+        assertEq(node.registerEpoch, 0);
+        assertTrue(
+            stakeTable._isEqualBlsKey(
+                node.blsVK,
+                BN254.G2Point(
+                    BN254.BaseField.wrap(0),
+                    BN254.BaseField.wrap(0),
+                    BN254.BaseField.wrap(0),
+                    BN254.BaseField.wrap(0)
+                )
+            )
+        );
+        assertTrue(EdOnBN254.isEqual(node.schnorrVK, EdOnBN254.EdOnBN254Point(0, 0)));
+
+        // look up the stake for the zero address and verify the data is empty
+        stakeAmount = stakeTable.lookupStake(address(0));
+        assertEq(stakeAmount, 0);
+
+        // look up the node for the zero address and verify the data is empty
+        node = stakeTable.lookupNode(address(0));
+        assertEq(node.account, address(0));
+        assertEq(node.balance, 0);
+        assertEq(node.registerEpoch, 0);
+        assertTrue(
+            stakeTable._isEqualBlsKey(
+                node.blsVK,
+                BN254.G2Point(
+                    BN254.BaseField.wrap(0),
+                    BN254.BaseField.wrap(0),
+                    BN254.BaseField.wrap(0),
+                    BN254.BaseField.wrap(0)
+                )
+            )
+        );
+        assertTrue(EdOnBN254.isEqual(node.schnorrVK, EdOnBN254.EdOnBN254Point(0, 0)));
+    }
+
+    function test_WithdrawFunds_succeeds() public {
```

Comment:
I would add another test where a user tries to withdraw twice. The first time it works, and then it reverts. I think it is ok to add this case in this same function.

Reply:
good one to add, done here https://github.com/EspressoSystems/espresso-sequencer/commit/f9bfe6dc4d821070d77aea3aecd7337b95f107f5